src/pages/registration/new_registration/LinkRegistration.tsx

```diff
@@ -1,67 +1,115 @@
+  [LinkRegistrationFormFieldName.Link]: string;
+}
+
+const emptyDoiFormValues: DoiFormValues = {
+  [LinkRegistrationFormFieldName.Link]: '',
+};
+
+const doiUrlBase = 'https://doi.org/';
+const doiUrlPlaceholder = `${doiUrlBase}10.1000/xyz123`;
+const doiRegExp = new RegExp('\\b(10[.][0-9]{4,}(?:[.][0-9]+)*/(?:(?!["&\'<>])\\S)+)\\b'); // https://stackoverflow.com/a/10324802
+
 export const LinkRegistration = ({ expanded, onChange }: StartRegistrationAccordionProps) => {
   const { t } = useTranslation();
-  const [doi, setDoi] = useState<Doi | null>(null);
-  const [noHit, setNoHit] = useState(false);
+  const [doiQuery, setDoiQuery] = useState('');
   const history = useHistory();
   const dispatch = useDispatch();
 
   const openRegistration = async () => {
-    if (!doi) {
+    if (!mutateDoi.data) {
       return;
     }
-    history.push(getRegistrationWizardPath(doi.identifier), { highestValidatedTab: -1 });
+    history.push(getRegistrationWizardPath(mutateDoi.data.data.identifier), { highestValidatedTab: -1 });
   };
 
-  const handleSearch = async (doiUrl: string) => {
-    setNoHit(false);
-    setDoi(null);
-
-    try {
-      const doiRegistrationResponse = await getRegistrationByDoi(doiUrl);
-      if (isErrorStatus(doiRegistrationResponse.status)) {
-        setNoHit(true);
-        dispatch(setNotification({ message: t('feedback.error.get_doi'), variant: 'error' }));
-      } else if (isSuccessStatus(doiRegistrationResponse.status)) {
-        if (doiRegistrationResponse.data) {
-          setDoi(doiRegistrationResponse.data);
-        } else {
-          setNoHit(true);
+  const resultsQuery = useQuery({
+    enabled: !!doiQuery,
+    queryKey: ['doi-results', doiQuery],
+    queryFn: () => fetchResults2(10, 0, `"${doiQuery}"`),
+    onSuccess: (response) => {
+      if (response.hits.length === 0) {
+        mutateDoi.mutate(doiQuery);
+      }
+    },
+  });
+
+  const searchResults = resultsQuery.data?.hits ?? [];
+
+  const mutateDoi = useMutation({
+    mutationFn: (doi: string) => getRegistrationByDoi(doi),
+    onSuccess: (response) => {
+      const doi = response.data;
+      if (doi) {
+        if (stringIncludesMathJax(doi.title)) {
+          typesetMathJax();
         }
       }
-    } catch {
-      setNoHit(true);
+    },
+    onError: () => {
       dispatch(setNotification({ message: t('feedback.error.get_doi'), variant: 'error' }));
-    }
-  };
+    },
+  });
+
+  const isLookingUpDoi = resultsQuery.isFetching || mutateDoi.isLoading;
+
+  const onSubmit = async (values: DoiFormValues, { setValues }: FormikHelpers<DoiFormValues>) => {
+    let doiUrl = values.link.trim().toLowerCase();
+    setDoiQuery(doiUrl);
 
-  useEffect(() => {
-    if (stringIncludesMathJax(doi?.title)) {
-      typesetMathJax();
+    if (!isValidUrl(doiUrl)) {
+      const regexMatch = doiRegExp.exec(doiUrl);
+      if (regexMatch && regexMatch.length > 0) {
+        doiUrl = `${doiUrlBase}${regexMatch[0]}`;
+      }
```

Comment:
Litt usikker på hvprdan man skal sende inn `values` til query om den ikke legges i state.

Reply:
prøvde å få til en løsning ved å kalle `.refetch()` herfra med noen argumenter, men var visst ikke så enkelt det heller. To mulige løsninger jeg ser:
- Droppe Formik, og bruke bare `useState` for å håndtere input.
- Flytte `<Formik>` ett hakk opp, slik at `useQuery` også får direkte tilgang til verdiene i formen

Men det er jo ikke noen krise slik det er nå heller da, så med litt tidsmangel her lener jeg mot å bare nøye oss med det sånn som det er